cloud/spawn.go

```diff
@@ -172,6 +178,35 @@ func CreateSpawnHost(so SpawnOptions) (*host.Host, error) {
 	return intentHost, nil
 }
 
+func CheckInstanceTypeValid(ctx context.Context, d distro.Distro, requestedType string, allowedTypes []string) error {
+	if !IsEc2Provider(d.Provider) {
+		return errors.Errorf("cannot specify instance type for provider '%s'", d.Provider)
+	}
+	foundType := false
+	for _, allowedType := range allowedTypes {
+		if requestedType == allowedType {
+			foundType = true
+			break
+		}
+	}
+	if !foundType {
+		return errors.New("This instance type has not been allowed by admins")
```

Comment:
Is this us admins or AWS admins?

Reply:
Us admins, I can specify that?

Reply:
I think it implies us. Wasn't sure if you meant AWS instead. LGTM as is.